src/caret_analyze/common/type_check_decorator.py

```diff
@@ -26,6 +26,62 @@
     from pydantic import ValidationError
     from pydantic.deprecated.decorator import validate_arguments
 
+    def _get_given_arg(
+        signature: Signature,
+        args: tuple[Any, ...],
+        kwargs: dict[str, Any],
+        given_arg_loc: tuple,
+        varargs: None | str
+    ) -> Any:
+        """
+        Get an argument which validation error occurs.
+
+        Parameters
+        ----------
+        signature: Signature
+            Signature of target function.
+        args: tuple[Any, ...]
+            Arguments of target function.
+        kwargs: dict[str, Any]
+            Keyword arguments of target function.
+        given_arg_loc: tuple
+            (i) Not iterable type case
+                ('<ARGUMENT_NAME>,')
+
+            (ii) Iterable type except for dict case
+                ('<ARGUMENT_NAME>', '<INDEX>')
+
+            (ii) Dict case
+                ('<ARGUMENT_NAME>', '<KEY>')
+        varargs: None | str
+            The name of the variable length argument if the function has one, otherwise None.
+
+        Returns
+        -------
+        str
+            The argument which validation error occurs.
+
+        """
+        arg_name = given_arg_loc[0]
+        given_arg: Any = None
+
+        # Check kwargs
+        for k, v in kwargs.items():
```

Comment:
I think to check kwargs can simple use the kwargs.get to get the given_arg.
`given_arg = kwargs.get(arg_name)`

Reply:
Thank you for thoughtful checking!

This change will be made in a later PR (PR which refactors type_check_decorator),
because this implementation was not conducted in this PR.